src/supertux/menu/sorted_contrib_menu.cpp

```diff
@@ -0,0 +1,84 @@
+//  SuperTux
+//  Copyright (C) 2021 mrkubax10 <mrkubax10@onet.pl>
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#include "supertux/menu/sorted_contrib_menu.hpp"
+
+#include <sstream>
+#include "util/gettext.hpp"
+#include "supertux/savegame.hpp"
+#include "supertux/player_status.hpp"
+#include "supertux/levelset.hpp"
+#include "supertux/game_manager.hpp"
+#include "supertux/menu/contrib_levelset_menu.hpp"
+#include "gui/menu_manager.hpp"
+#include "gui/menu_item.hpp"
+#include "gui/item_action.hpp"
+SortedContribMenu::SortedContribMenu(std::vector<std::unique_ptr<World>>& worlds, std::string contrib_type, std::string title, std::string empty_message) : 
+  m_world_folders()
+{ 
+  add_label(title);
+  add_hl();
+  int world_id = 0;
+  for (unsigned int i = 0; i < worlds.size(); i++)
+  {
+    if (worlds[i]->get_contrib_type() == contrib_type)
+    {
+      auto savegame = Savegame::from_file(worlds[i]->get_savegame_filename());
+      if (worlds[i]->is_levelset())
+      {
+        m_world_folders.push_back(worlds[i]->get_basedir());
+        std::ostringstream title_stream;
+        title_stream << "[" << worlds[i]->get_title() << "]";
+        std::ostringstream desc;
+        desc << worlds[i]->get_description();
+        add_entry(world_id++, title_stream.str()).set_help(desc.str());
+      }
+      else if (worlds[i]->is_worldmap())
+      {
+        m_world_folders.push_back(worlds[i]->get_basedir());
+        std::ostringstream title_stream;
+        title_stream << worlds[i]->get_title();
+        std::ostringstream desc;
+        desc << worlds[i]->get_description();
+        add_entry(world_id++, title_stream.str()).set_help(desc.str());
+      }
```

Comment:
There's a lot of similarities in both parts of the `if`; isn't it possible to move some of it outside?

Possibly use the `?` operator if there isn't too many differences.